app/lib/service/openid/jwt.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+/// Pattern for a valid JWT, these must 3 base64 segments separated by dots.
+final _jwtPattern = RegExp(
+    r'^[a-zA-Z0-9+/=_-]{4,}\.[a-zA-Z0-9+/=_-]{4,}\.[a-zA-Z0-9+/=_-]{4,}$');
```

Comment:
https://datatracker.ietf.org/doc/html/rfc7519#section-2
says:
> The terms "JSON Web Signature ([JWS](https://datatracker.ietf.org/doc/html/rfc7519#ref-JWS))", "Base64url Encoding", "Header
   Parameter", "JOSE Header", "JWS Compact Serialization", "JWS
   Payload", "JWS Signature", and "Unsecured JWS" are defined by the JWS
   specification [JWS].

JWS, [RFC7515 Appendix C specifies](https://www.rfc-editor.org/rfc/rfc7515.html#appendix-C)
>    This appendix describes how to implement base64url encoding and
   decoding functions without padding based upon standard base64
   encoding and decoding functions that do use padding.

Hence, as far as I can see padding is not allowed.

-------------

Please consider including references to these RFCs in source code comment.